src/fides/api/ops/service/connectors/erasure_email_connector.py

```diff
@@ -0,0 +1,201 @@
+            if filtered_user_identities:
+                batched_identities.extend(filtered_user_identities.values())
+            else:
+                skipped_privacy_requests.append(privacy_request.id)
+
+        if not batched_identities:
+            logger.info(
+                "Skipping erasure email send for connector: '{}'. "
+                "No corresponding user identities found for pending privacy requests.",
+                self.configuration.name,
+            )
+            return
+
+        logger.info(
+            "Sending batched erasure email for connector {}...",
+            self.configuration.name,
+        )
+
+        db = Session.object_session(self.configuration)
+
+        try:
+            send_single_erasure_email(
+                db=db,
+                subject_email=self.config.recipient_email_address,
+                subject_name=self.config.third_party_vendor_name,
+                batch_identities=batched_identities,
+                test_mode=False,
+            )
+        except MessageDispatchException as exc:
+            logger.info("Erasure email failed with exception {}", exc)
+            raise
+
+        # create an audit event for each privacy request ID
+        for privacy_request in privacy_requests:
+            ExecutionLog.create(
+                db=db,
+                data={
+                    "connection_key": self.configuration.key,
+                    "dataset_name": self.configuration.name,
+                    "privacy_request_id": privacy_request.id,
+                    "action_type": ActionType.erasure,
+                    "status": ExecutionLogStatus.complete,
+                    "message": f"Erasure email instructions dispatched for {self.config.third_party_vendor_name}",
+                },
+            )
+
+
+def get_identity_types_for_connector(
+    email_secrets: EmailSchema,
+) -> List[str]:
+    """Return a list of identity types we need to email to the third party vendor."""
+    advanced_settings: AdvancedSettings = email_secrets.advanced_settings
+    identity_types: IdentityTypes = advanced_settings.identity_types
+    flattened_list: List[str] = []
+
+    if identity_types.email:
+        flattened_list.append("email")
+    if identity_types.phone_number:
+        flattened_list.append("phone_number")
+
+    return flattened_list
+
+
+def filter_user_identities_for_connector(
+    secrets: EmailSchema, user_identities: Dict[str, Any]
+) -> Dict[str, Any]:
+    """Filter identities to just those specified for a given connector"""
+    required_identities: List[str] = get_identity_types_for_connector(secrets)
+    return {
+        identity_type: user_identities.get(identity_type)
+        for identity_type in required_identities
+        if user_identities.get(identity_type)
+    }
+
+
+def send_single_erasure_email(
+    db: Session,
+    subject_email: str,
+    subject_name: str,
+    batch_identities: List[str],
```

Comment:
I like the naming of this to help call out it's one single email with potentially multiple identities